src/api/apiMethods.js

```diff
@@ -0,0 +1,235 @@
+/**
+ * Copyright Schrodinger, LLC
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+import _ from 'lodash';
+import columnWidths from '../selectors/columnWidths';
+import shallowEqualSelector from '../helper/shallowEqualSelector';
+
+/**
+ * Minimal data of a column that we expose through our APIs
+ */
+const _getMinimalColumn = (column) => ({
+  offset: column.offset,
+  width: column.width,
+  index: column.index,
+  columnKey: column.columnKey,
+});
+
+/**
+ * Minimal data of a column group that we expose through our APIs
+ */
+const _getMinimalColumnGroup = (columnGroup) => ({
+  offset: columnGroup.offset,
+  width: columnGroup.width,
+  index: columnGroup.index,
+  columnKey: columnGroup.columnKey,
+});
+
+/**
+ * Return all exposed APIs
+ *
+ * @param {!Object} state
+ * @param {!Object} api
+ * @returns
+ */
+const getApiMethodsSelector = () =>
+  shallowEqualSelector(
```

Comment:
This returns the public method APIs that we expose to users.
This is also memoized for efficiency.